src/jobs/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { Application } from '../domain/interface';
+import logger from '../logger';
+import { Report, ReportItem } from './types';
+
+export const getEmptyReport: () => ReportItem = () => ({
+  count: 0,
+  ids: [],
+  errors: [],
+  errorCount: 0,
+});
+
+export const buildReportItem = (
+  results: PromiseSettledResult<any>[],
+  reportType: keyof Report,
+  jobName: string,
+) => {
```

Comment:
Declare the function return type here so we can be assured that it always returns the expected type. Taking advantage of the TypeScript compiler for our benefit.
```suggestion
export const buildReportItem = (
  results: PromiseSettledResult<any>[],
  reportType: keyof Report,
  jobName: string,
): ReportItem => {
```